webapp/src/main/java/info/teksol/mindcode/webapp/CompileMain.java

```diff
@@ -0,0 +1,70 @@
+package info.teksol.mindcode.webapp;
+
+import info.teksol.mindcode.Tuple2;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.util.List;
+
+import static info.teksol.mindcode.webapp.CompilerFacade.compile;
+
+public class CompileMain {
+    static private String readFile(String filename) throws IOException {
+        try (final BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+            final StringWriter out = new StringWriter();
+            reader.transferTo(out);
+            return out.toString();
+        }
+    }
+
+    static private boolean compileMindcode(String contents, boolean optimise) {
+        final Tuple2<String, List<String>> result = compile(contents, optimise);
+
+        final String compiledCode = result._1;
+        final List<String> syntaxErrors = result._2;
+
+        if (!syntaxErrors.isEmpty()) {
+            // Print errors to stderr.
+            for (int i = 0; i < syntaxErrors.size(); i++) {
+                System.err.println(syntaxErrors.get(i));
+            }
+            return false;
+        } else {
+            // No errors? Print the compiled code.
+            System.out.println(compiledCode);
+            return true;
+        }
```

Comment:
Nitpick: this would be easier to read / follow if the two branches were reversed. I always found positive conditionals to be easier to read. In this case, I would have done:

```java
if (syntaxErrors.isEmpty() {
  // print
} else {
  // handle errors
}
```